SCF/SCFSession6/IntSet/IntSet.java

```diff
@@ -0,0 +1,183 @@
+package GET2018.SCF.SCFSession6.IntSet;
+
+/**
+ * This Immutable class Performs various Operations on Set
+ * 
+ * @author Mohit Sharma
+ *
+ */
+public final class IntSet {
+	static final int intSetRange[] = new int[1000];
+	final int intSet[];
+
+	IntSet(int intSetInput[]) throws SetException {
+		if (intSetInput == null) {
+			throw new NullPointerException("Input Array is Null");
+		}
+		if (intSetInput.length == 0) {
+			throw new SetException("Input Array Length can't be Zero");
+		}
+		int checkMembership;
+		for (int i = 0; i < intSetRange.length; i++) {
+			intSetRange[i] = i + 1;
+		}
+
+		/*
+		 * Checks Whether the values of Array are in Range 1-1000 or not
+		 */
+		for (int i = 0; i < intSetInput.length; i++) {
+			for (checkMembership = 0; checkMembership < intSetRange.length; checkMembership++) {
+				if (intSetInput[i] == intSetRange[checkMembership]) {
+					break;
+				}
+			}
+			if (checkMembership == intSetRange.length) {
+				throw new SetException("Input Invalid Not in Range 1-1000");
+			}
+		}
+		IntSet.sortArray(intSetInput);
+		intSet = intSetInput;
+	}
+
+	/**
+	 * Check whether x is a member of the Set
+	 * 
+	 * @param x
+	 * @return
+	 */
+	public boolean isMember(int x) {
+		for (int i = 0; i < intSetRange.length; i++) {
+			if (x == intSetRange[i]) {
+				return true;
+			}
+		}
+		return false;
+	}
+
+	/**
+	 * Return the size of the Set
+	 * 
+	 * @return
+	 */
+	public int size() {
+		return intSet.length;
+	}
+
+	/**
+	 * Check whether subsetSet is a Subset of the Set
+	 * 
+	 * @param set
+	 * @return
+	 */
+	public boolean isSubSet(IntSet subsetSet) {
+		int checkMembership;
+		for (int i = 0; i < subsetSet.intSet.length; i++) {
+			for (checkMembership = 0; checkMembership < this.intSet.length; checkMembership++) {
+				if (this.intSet[checkMembership] == subsetSet.intSet[i]) {
+					break;
+				}
+			}
+			if (checkMembership == this.intSet.length) {
+				return false;
+			}
+		}
+		return true;
+	}
+
+	/**
+	 * Return the complement set
+	 */
+	public void getComplement() {
+		int checkMembership;
+		int complementSet[] = new int[1000];
+		int indexOfComplementSet = 0;
+		for (int i = 0; i < intSetRange.length; i++) {
+			for (checkMembership = 0; checkMembership < intSet.length; checkMembership++) {
+				if (intSetRange[i] == intSet[checkMembership]) {
+					break;
+				}
+			}
+			if (checkMembership == intSet.length) {
+				complementSet[indexOfComplementSet] = intSetRange[i];
+				indexOfComplementSet++;
+			}
+		}
+		for (int x = 0; x < indexOfComplementSet; x++) {
+			System.out.println(complementSet[x]);
+		}
+	}
+
+	/**
+	 * Return the Union of set1 and set2
+	 * 
+	 * @param set1
+	 * @param set2
+	 * @return
+	 */
+	public static int[] union(IntSet set1, IntSet set2) {
+		int unionSet[] = new int[1000];
```

Comment:
I think we can do this without using 3rd set. Please optimize this logic.

Reply:
The size of both sets set1 and set2 are fixed so we can't add element to these.
Therefore, we have to take a extra Set for storing all the Union Elements.